src/main/java/cinema/project/service/AuthenticationServiceImpl.java

```diff
@@ -29,6 +31,8 @@ public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
-        return userService.add(user);
+        User user1 = userService.add(user);
+        shoppingCartService.registerNewShoppingCart(user1);
+        return user1;
```

Comment:
you can use the same user reference even without reassigning it. Check whether user object will have an id after userService.add()